python/cuml/dask/common/input_utils.py

```diff
@@ -215,11 +215,12 @@ def _extract_partitions(dask_obj, client=None):
     # dask.dataframe or dask.array
     if isinstance(dask_obj, dcDataFrame) or \
             isinstance(dask_obj, daskArray):
-        parts = futures_of(client.compute(dask_obj))
+        parts = client.compute(futures_of(client.persist(dask_obj)))
 
     # iterable of dask collections (need to colocate them)
     elif isinstance(dask_obj, Sequence):
-        parts = [futures_of(client.compute(a)) for a in dask_obj]
+        parts = [client.compute(futures_of(client.persist(d)))
+                 for d in dask_obj]
```

Comment:
What about persisting `dask_obj` before the `if` branches? This would allow all branches to benefit. Also Dask could benefit from the additional context about all things being computed.

Reply:
Sure. What would the benefit be from the early persists?

Reply:
I believe rather than having to persist in multiple places, you can just persist once.

Reply:
Also if the `d`s in `dask_obj` share a common subset of the graph, Dask can leverage that information to schedule things more efficiently when computing all of them.